src/python/pants/backend/python/lint/ruff/rules_integration_test.py

```diff
@@ -198,7 +198,7 @@ def test_skip_field(rule_runner: RuleRunner) -> None:
 
     fix_result, lint_result, fmt_result = run_ruff(rule_runner, tgts)
 
-    assert lint_result.exit_code == 1
+    assert lint_result.exit_code == 0
```

Comment:
I haven't yet done any investigation to establish why this behaviour changes, but should

Reply:
Ah, the old behaviour was incorrect. The new behaviour is correct. This test ends up invoking `ruff check` with _no_ file arguments (because they're all skipped), which means ruff does its own directory walking to find what to lint.

- Previously, when using a venv pex: there were a few python files within the ruff pex included in the input digest/snapshot, that the auto-traversal was finding: `ruff.pex/__main__.py`, `ruff.pex/__pex__/__init__.py`. These files fail linting (e.g. ``ruff.pex/__main__.py:12:24: F401 [*] `typing.Any` imported but unused``), and thus the lint `ruff check` invocation fails.
- Now, when using the binary directory: there's no python files in the sandbox, so the auto-traversal finds nothing and thus the `ruff check` invocation succeeds (although has a warning about not finding any Python files)

So, calling this good enough ✅